pkg/controller/egress_router/egress_router_controller.go

```diff
@@ -0,0 +1,159 @@
+package egress_router
+
+// egress router imnplements a controller for the egress router CNI plugin
+
+import (
+	"context"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/certrotation"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"reflect"
+	"log"
+	"strings"
+	"time"
+
+	netopv1 "github.com/openshift/cluster-network-operator/pkg/apis/network/v1"
+	"github.com/openshift/cluster-network-operator/pkg/controller/statusmanager"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// Attach control loop to the manager and watch for Egress Router objects
+func Add(mgr manager.Manager, status *statusmanager.StatusManager) error {
+	r, err := newEgressRouterReconciler(mgr, status)
+	if err != nil {
+		return err
+	}
+
+	// Create a new controller
+	c, err := controller.New("egress-router-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource EgressRouter.network.operator.openshift.io/v1
+	err = c.Watch(&source.Kind{Type: &netopv1.EgressRouter{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &EgressRouterReconciler{}
+
+type EgressRouterReconciler struct {
+	mgr       manager.Manager
+	clientset *kubernetes.Clientset
+	status    *statusmanager.StatusManager
+
+	egressrouters map[types.NamespacedName]*egressrouter
+	egressrouterErrs map[types.NamespacedName]error
+}
+
+var ResyncPeriod = 5 * time.Minute
```

Comment:
I don't think this needs to be public? And if it needs to be public it should have a description

Reply:
this is also used like that in the PKIcontroller, take a look ;)